RockWeb/Blocks/Reporting/DataViewDetail.ascx.cs

```diff
@@ -142,6 +142,47 @@ protected void btnEdit_Click( object sender, EventArgs e )
         }
 
         /// <summary>
+        /// Handles the Click event of the Copy button control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs" /> instance containing the event data.</param>
+        protected void btnCopy_Click( object sender, EventArgs e )
+        {
+            // Create a new Data View using the current item as a template.
+            var id = int.Parse( hfDataViewId.Value );
+            
+            var service = new DataViewService( new RockContext() );
```

Comment:
@MrUpsideDown I'd suggest naming this variable what it is (dataViewService).  Specific variable names are easier to maintain and read than generic ones.